lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -872,4 +867,35 @@ previewers.display_content = defaulter(function(_)
   }
 end, {})
 
+previewers.buffers = defaulter(function(opts)
+  opts = opts or {}
+  local cwd = opts.cwd or vim.loop.cwd()
+  return Previewer:new {
+    title = function()
+      return "Buffers"
+    end,
+    setup = function(_, status)
+      return { ["winid"] = status.preview_win }
+    end,
```

Comment:
We can just set `self.state.winid` in `preview_fn` every time. I dont really see a reason to give setup access to status

Reply:
I've actually just refactored, such that `setup` and `teardown` take care of setting the window options (as opposed to in `builtin.buffers`) accordingly for which I would need status to get `original_win_id`. Fine to keep status here then?

Reply:
then yeah